Library/Homebrew/dependency.rb

```diff
@@ -124,11 +124,39 @@ def merge_repeats(all)
       all.map(&:name).uniq.map do |name|
         deps = grouped.fetch(name)
         dep  = deps.first
-        tags = deps.flat_map(&:tags).uniq
+        tags = merge_tags(deps)
         option_names = deps.flat_map(&:option_names).uniq
         dep.class.new(name, tags, dep.env_proc, option_names)
       end
     end
+
+    private
+
+    def merge_tags(deps)
+      options = deps.flat_map(&:option_tags).uniq
+      merge_necessity(deps) + merge_temporality(deps) + options
+    end
+
+    def merge_necessity(deps)
+      if deps.any? { |dep| !dep.recommended? && !dep.optional? }
+        # FIXME: Should be `required?`, but `required?` implies `!build?`.
```

Comment:
Yes, maybe the `FIXME` should go into the definition of `required?`.

Ideally, our treatment of dependencies would be a bit more uniform than is currently the case (if my assessment of the current situation is correct), resulting in a different definition of `required?` or both `optional?` and `recommended?` in `Dependable`. But that would be a too far-reaching change for this PR (and it requires substantially more testing). This PR on the other hand tries to be a fix that can be merged (hopefully) relatively quickly.